soc/arm64/intel_socfpga/agilex/mmu_regions.c

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright (c) 2021, Intel Corporation. All rights reserved.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <soc.h>
+#include <arch/arm64/arm_mmu.h>
+
+static const struct arm_mmu_region mmu_regions[] = {
+
+	MMU_REGION_FLAT_ENTRY("PERIPHERAL",
+			      0xF7000000,
+			      0x08E00000,
+			      MT_DEVICE_nGnRnE | MT_RW | MT_SECURE),
+
+	MMU_REGION_FLAT_ENTRY("MEMORY",
+			      DT_REG_ADDR(DT_NODELABEL(mem0)),
+			      DT_REG_SIZE(DT_NODELABEL(mem0)),
+			      MT_NORMAL | MT_RW | MT_SECURE),
+
+	MMU_REGION_FLAT_ENTRY("GIC",
+			      DT_REG_ADDR_BY_IDX(DT_NODELABEL(gic), 0),
+			      DT_REG_SIZE_BY_IDX(DT_NODELABEL(gic), 0),
+			      MT_DEVICE_nGnRnE | MT_RW | MT_SECURE),
```

Comment:
Are you sure you want this to be `MT_SECURE`?

Reply:
I have changed it to MT_NS, it doesn't need to be secure.

Reply:
Not sure about your use-case but you can think to use `MT_DEFAULT_SECURE_STATE`

Reply:
Noted, I will change to use MT_DEFAULT_SECURE_STATE. And, will adjust accordingly in future when needed.